app/api/api_v1/endpoints/submission_router.py

```diff
@@ -29,13 +22,36 @@ def create_submission(
         raise HTTPException(status_code=423, detail="Assignment has not been released")
     if assignment.get_is_closed_for_student(db, onyen):
         raise HTTPException(status_code=423, detail="Assignment is closed for submission")
+    
+    return student, assignment
+
+@router.post("/submission/", response_model=SubmissionSchema)
+def create_submission(
+    *,
+    db: Session = Depends(get_db),
+    onyen: int,
+    assignment_id: int,
+    commit_id: str
+):
+    student, assignment = validate_submission(db, onyen, assignment_id)
     submission = SubmissionModel(
         student_id=student.id,
-        assignment_id=assignment_id,
+        assignment_id=assignment.id,
         commit_id=commit_id
     )
     
     db.add(submission)
     db.commit()
 
-    return submission
\ No newline at end of file
+    return submission
+
+@router.get("/submission/", response_model=Page[SubmissionSchema])
+def get_submission(
+    *,
+    db: Session = Depends(get_db),
+    onyen: int,
+    assignment_id: int
+):
+    student, assignment = validate_submission(db, onyen, assignment_id)
+    submission = db.query(SubmissionModel).filter_by(student_id=student.id, assignment_id=assignment.id, submission_time=assignment.get_latest_submission_time(db, onyen)).first()
```

Comment:
```suggestion
    onyen: int,
    assignment_id: int
):
    student, assignment = validate_submission(db, onyen, assignment_id)
    submission = db.query(SubmissionModel)
                    .filter_by(
                        student_id=student.id, 
                        assignment_id=assignment.id, 
                        submission_time=assignment.get_latest_submission_time(db, onyen))
                    .first()
```suggestion
    onyen: int,
    assignment_id: int
):
    student, assignment = validate_submission(db, onyen, assignment_id)
    submission = db.query(SubmissionModel).filter_by(student_id=student.id, assignment_id=assignment.id, submission_time=assignment.get_latest_submission_time(db, onyen)).first()
```
```